sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -921,7 +921,9 @@ class Analyzer(override val catalogManager: CatalogManager)
     }
 
     def lookupTempView(
-        identifier: Seq[String], isStreaming: Boolean = false): Option[LogicalPlan] = {
+        identifier: Seq[String],
+        isStreaming: Boolean = false,
+        performCheck: Boolean = false): Option[LogicalPlan] = {
```

Comment:
when shall we skip the check?

Reply:
For `UnresolvedView`, `UnresolvedTable`, `UnresolvedTableOrView`, we use `lookupTempView` to check whether the temp view exists, but it's ok to skip the check. For example:
1. In DropView, the UnresolvedView will change to ResolvedView even the referred table is dropped.
2. When resolving `UnresolvedTable`, if it is a view, should analyzer should throw `expectTableNotViewError` rather than `table or view not found`

Reply:
@imback82 This is something we should improve. For `SELECT ... FROM view` and `DROP VIEW view`, the way to lookup the view should be different.

For `SELECT ... FROM view`, we must fully resolve the view, as we need to execute it.

For `DROP VIEW view`, we only need to get the view metadata entry. The same to `ALTER VIEW ...`.

I think we should change how we resolve `UnresolvedTableOrView` and `UnresolvedView`, to only get the view metadata.

Reply:
OK, I will think about this.